### PR TITLE
Add SDK storybook PoC

### PR DESCRIPTION
**Pull request by deniskaber**
Relates to https://www.notion.so/metabase/Embedding-SDK-e2e-tests-c3c652dd4c5b452a9a243266ae434bb3

### Description

This adds storybook and e2e tests for Embedding SDK.
Storybook acts as an example host app to show Embedding SDK components.
Then we run e2e tests to open storybook story and automate a user flow.

### How to verify

Storybook:
- `yarn dev-ee` to run Metabase.
- Go to `http://localhost:3000/admin/settings/authentication/jwt` and setup JWT secret to be "0000000000000000000000000000000000000000000000000000000000000000"
- So to `http://localhost:3000/admin/settings/embedding-in-other-applications/full-app` and set AUTHORIZED ORIGINS to "`*`"
- `yarn storybook-embedding-sdk`
- Open `http://localhost:6006/?path=/story/embeddingsdk-staticdashboard--default`

e2e tests:
Local:
- Run storybook from the previous step
- Run tests locally with `yarn test-cypress-open-sdk`
- `e2e/test/scenarios/embedding-sdk/static-dashboard.cy.spec.js` scenario should pass

Screenshot 2024-07-19 at 20 09 55

CI:
`e2e-tests-embedding-sdk-ee` CI job should pass

### Demo

Screenshot 2024-07-13 at 00 06 41

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Review by heypoom — APPROVED** [review]
Nice work! Implementation looks good to me.

**Review by nemanjaglumac — CHANGES_REQUESTED** [review]
Left a lot of inline comments, but I assume this might be faster resolved in a quick call.

File: .github/actions/run-embedding-sdk-storybook/action.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Run Embedding SDK Storybook
```
Comment (nemanjaglumac) [inline .github/actions/run-embedding-sdk-storybook/action.yml:1]: NIt: I would rename this workflow (both its name and file on disk) to `prepare-embedding-sdk-environment`.

1. we're not actually running tests in this action
2. we might find some solution other than the Storybook in the future

File: .github/actions/run-embedding-sdk-storybook/action.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+runs:
+  using: "composite"
+  steps:
+    - name: Prepare frontend environment
+      uses: ./.github/actions/prepare-frontend
+
+    - name: Prepare back-end environment
+      uses: ./.github/actions/prepare-backend
+      with:
+        m2-cache-key: "cljs"
```
Comment (nemanjaglumac) [inline .github/actions/run-embedding-sdk-storybook/action.yml:11]: What was the issue without prepare-frontend and prepare-backend in here?

File: e2e/test/scenarios/embedding-sdk/static-dashboard.cy.spec.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  });
+
+  it("should show dashboard content", () => {
+    cy.get("@dashboardId").then(dashboardId => {
+      visitFullAppEmbeddingUrl({
+        url: EMBEDDING_SDK_STORY_HOST,
+        qs: { id: "embeddingsdk-staticdashboard--default", viewMode: "story" },
+        onBeforeLoad: window => {
+          window.JWT_SHARED_SECRET = JWT_SHARED_SECRET;
+          window.METABASE_INSTANCE_URL = "http://localhost:4000";
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/embedding-sdk/static-dashboard.cy.spec.js:59]: Please do not hard code Cypress url.

```suggestion
          window.METABASE_INSTANCE_URL = Cypress.config().baseUrl;
```

File: e2e/test/scenarios/embedding-sdk/static-dashboard.cy.spec.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      expect(response?.statusCode).to.equal(200);
+    });
+
+    cy.get("#metabase-sdk-root")
+      .should("be.visible")
+      .within(() => {
+        cy.findByText("Embedding Sdk Test Dashboard").should("be.visible"); // dashboard title
+
+        cy.findByText("Orders").should("be.visible"); // question card content
+        cy.findByText("Text card").should("be.visible"); // text card content
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/embedding-sdk/static-dashboard.cy.spec.js:79]: If card's test-id is available, can we please scope this to that?
I'm afraid simply finding the text "Orders" will backfire in the future.

File: enterprise/frontend/src/embedding-sdk/test/common-stories-utils.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+const DEFAULT_CONFIG: SDKConfig = {
+  metabaseInstanceUrl: METABASE_INSTANCE_URL,
+  jwtProviderUri: `${METABASE_INSTANCE_URL}/sso/metabase`,
+  fetchRequestToken: async () => {
+    try {
+      const signedUserData = await new jose.SignJWT({
+        email: DEFAULT_SDK_USER.email,
+        first_name: DEFAULT_SDK_USER.firstName,
+        last_name: DEFAULT_SDK_USER.lastName,
```
Comment (nemanjaglumac) [inline enterprise/frontend/src/embedding-sdk/test/common-stories-utils.tsx:28]: Two questions:

1. Couldn't we reuse the `normal` user from `cypress_data.js`?
2. What happens if first_name and/or last_name are not present? They are NOT the requirement in the main application - only email is.

**Review by nemanjaglumac — APPROVED** [review]
Added a few more comments.
Please address https://github.com/metabase/metabase/pull/45512/files#r1689230096 before merging.

Great work! 🚀

File: .github/actions/run-embedding-sdk-storybook/action.yml (hunk 2 above)
Comment (nemanjaglumac) [reply .github/actions/run-embedding-sdk-storybook/action.yml:11]: I understand that part, but we already prepare both frontend and backend in this job.

**Pull request merged**